src/soar_sdk/abstract.py

```diff
@@ -207,11 +194,33 @@ def delete(
         return response
 
     def get_soar_base_url(self) -> str:
+        """Gets the base URL for the SOAR API."""
         return "https://localhost:9999/"
 
     @abstractmethod
     def update_client(self, soar_auth: SOARClientAuth, asset_id: str) -> None:
+        """Updates the client before an actions run with the input data.
+
+        An example of what this function might do is authenticate the API client.
         """
-        Updates the client before an actions run with the input data. An example of what this function might do is authenticate the api client.
-        """
+        pass
+
+    @abstractmethod
+    def set_summary(self, summary: SummaryType) -> None:
+        """Sets the summary for the action run."""
+        pass
+
+    @abstractmethod
+    def set_message(self, message: str) -> None:
+        """Sets the summary message for the action run."""
```

Comment:
We should aim for a consistent tense in docs. Not sure which has more inertia across the whole project, but the code that existed in this file previously was like this:
```suggestion
        """Set the summary message for the action run."""
```